### ISSUE TITLE
Recursion in intersection

### ISSUE THREAD

**Issue by uittorio** [uittorio/ts-auto-mock#90]
This Interface should be compiled correctly without giving errors

```ts
interface InterfaceWithIntersection {
    int: InterfaceWithIntersection & B;
 }

 createMock<InterfaceWithIntersection>

```
This interface is valid because an intersection (InterfaceWithIntersection & B) its just the combination of multiple types.
The current compiler is failing because is trying to resolve all the properties of an intersection type causing a maximum call stack error. 

_This is what is happening_
 1) It will first resolve Interface
 2) it will find the property int
 3) it will detect that is an InterfaceWithIntersection
 4) it will resolve all the properties of InterfaceWithIntersection & B.

Because its resolving all the properties of InterfaceWithIntersection it will do keep resolving itself.

The proposed solution is to create a cache declaration for Intersection type, like we do for any other declarations.

The cache declaration will contains 2 declarations and every time the compiler will find an intersectionType it will check the cache first, so it will not resolve all the properties again

With this solution we have to be careful about
 - generics,
 - intersection between primitive types (string)
 - intersection with typescript libraries, like promises

We should create multiple tests for all the scenario above

**Comment by uittorio** [comment]
#83

**Comment by uittorio** [comment]
this issue is resolved by #93

**Comment by uittorio** [comment]
This issue has been resolved in version 1.3.1
